### PR TITLE
Display review state of each learning goal

### PR DESCRIPTION
**Pull request by Nokondi**
Modified LearningGoal header to display the following:

- "Evaluate" when AI is not enabled, teacher editing is enabled, and no understanding is selected, 
- "Approve" when AI is enabled, teacher editing is enabled, and no understanding is selected,
- Current understanding when understanding is selected,
- Nothing when no understanding is selected and teacher editing is disabled.

Screenshot 2023-10-04 095147

Screenshot 2023-10-04 095157

##Follow up
Could not complete unit tests of understanding behaviors due to enzyme's inability to handle hooks in functional components.

### CODE REVIEW

File: apps/src/templates/rubrics/LearningGoal.jsx [hunk 1]
```diff
@@ -29,19 +30,19 @@ export default function LearningGoal({
...
 }) {
   const [isOpen, setIsOpen] = useState(false);
-
   const [isAutosaving, setIsAutosaving] = useState(false);
   const [autosaved, setAutosaved] = useState(false);
   const [errorAutosaving, setErrorAutosaving] = useState(false);
   const [learningGoalEval, setLearningGoalEval] = useState(null);
   const [displayFeedback, setDisplayFeedback] = useState('');
-  const understandingLevel = useRef(0);
+  const [displayUnderstanding, setDisplayUnderstanding] = useState(-1);
```
Comment (bethanyaconnor) [inline apps/src/templates/rubrics/LearningGoal.jsx]: nit: can default to `null` instead? I think it makes the conditionals below clearer if we can use `!displayUnderstanding`

File: apps/src/templates/rubrics/LearningGoal.jsx [hunk 2]
```diff
@@ -107,15 +108,17 @@ export default function LearningGoal({
...
             setDisplayFeedback(teacherFeedback.current);
           }
-          if (json.understanding) {
+          if (json.understanding >= 0 && json.understanding !== null) {
             understandingLevel.current = json.understanding;
+            setDisplayUnderstanding(json.understanding);
+          } else {
+            setDisplayUnderstanding(-1);
           }
+          console.log(understandingLevel.current);
```
Comment (bethanyaconnor) [inline apps/src/templates/rubrics/LearningGoal.jsx]: nit: remove console log

File: apps/src/templates/rubrics/LearningGoal.jsx [hunk 3]
```diff
@@ -153,8 +157,20 @@ export default function LearningGoal({
         <div className={style.learningGoalHeaderRightSide}>
           {aiEnabled && <AiToken />}
           {/*TODO: Display status of feedback*/}
-          {canProvideFeedback && (
-            <BodyThreeText>{i18n.needsApproval()}</BodyThreeText>
+          {canProvideFeedback &&
+            aiEnabled &&
+            understandingLevel.current === -1 && (
```
Comment (bethanyaconnor) [inline apps/src/templates/rubrics/LearningGoal.jsx]: I think this should use `displayUnderstanding` instead, right? Maybe asked a different way: why use the ref over the state here?

File: apps/test/unit/templates/rubrics/EvidenceLevelsTest.jsx [hunk 4]
```diff
@@ -30,4 +31,17 @@ describe('EvidenceLevels', () => {
     // Two BodyThreeText per evidence level
     expect(wrapper.find('BodyThreeText').length).to.equal(4);
   });
+
+  it('calls radioButtonCallback when understanding is selected', () => {
+    const callback = sinon.stub();
+    const wrapper = mount(
```
Comment (bethanyaconnor) [inline apps/test/unit/templates/rubrics/EvidenceLevelsTest.jsx:64]: can this use shallow over mount? (it may not be able to, not sure)

File: apps/test/unit/templates/rubrics/LearningGoalTest.jsx [hunk 5]
```diff
@@ -129,4 +129,38 @@ describe('LearningGoal', () => {
...
+          aiEnabled: false,
+          evidenceLevels: [],
+        }}
+        teacherHasEnabledAi
+        canProvideFeedback
+      />
+    );
+    wrapper.update();
+    expect(wrapper.find('BodyThreeText').text()).to.include('Evaluate');
+    wrapper.unmount();
```
Comment (bethanyaconnor) [inline apps/test/unit/templates/rubrics/LearningGoalTest.jsx:190]: out of curiosity, why is this needed?

File: apps/src/templates/rubrics/LearningGoal.jsx (hunk 1 above)
Comment (Nokondi) [reply apps/src/templates/rubrics/LearningGoal.jsx]: I'm using -1 because I was having an issue where the conditionals were recognizing 0 as a false value, but 0 is the value for "No Evidence," so it wasn't displaying whenever No Evidence was selected.

File: apps/src/templates/rubrics/LearningGoal.jsx (hunk 3 above)
Comment (Nokondi) [reply apps/src/templates/rubrics/LearningGoal.jsx]: I removed the understandingLevel ref completely, and it seems to be working fine. Had included that because I was worried about the value not updating until re-render, but it seems to be ok.

File: apps/src/templates/rubrics/LearningGoal.jsx (hunk 1 above)
Comment (bethanyaconnor) [reply apps/src/templates/rubrics/LearningGoal.jsx]: ah got it. Can you extract the -1 into a constant? (more of a nit than a real concern)

File: apps/test/unit/templates/rubrics/EvidenceLevelsTest.jsx (hunk 4 above)
Comment (Nokondi) [reply apps/test/unit/templates/rubrics/EvidenceLevelsTest.jsx:64]: It has to be mount for the function stub to work.

File: apps/test/unit/templates/rubrics/LearningGoalTest.jsx (hunk 5 above)
Comment (Nokondi) [reply apps/test/unit/templates/rubrics/LearningGoalTest.jsx:190]: I noticed in the enzyme documentation that you should always unmount after using the mount command to prevent memory leaks.

Comment (bethanyaconnor) [reply apps/test/unit/templates/rubrics/LearningGoalTest.jsx:190]: huh TIL!

**Review by bethanyaconnor — APPROVED** [review]
LGTM! I got one nit comment in there (possibly between iterations) but other than that looks great!

File: apps/test/unit/templates/rubrics/LearningGoalTest.jsx (hunk 5 above)
Comment (davidsbailey) [reply apps/test/unit/templates/rubrics/LearningGoalTest.jsx:190]: yes, nice find!
